python/pyarrow/types.pxi

```diff
@@ -149,14 +154,14 @@ def _is_primitive(Type type):
 
 def _get_pandas_type(arrow_type, coerce_to_ns=False):
     cdef Type type_id = arrow_type.id
-    if type_id not in _pandas_type_map:
+    if type_id not in _get_pandas_type_map():
```

Comment:
Will this recreate the dict everytime this function is called? We don't want this.

Reply:
Might `functools.cache` be enough here? To avoid dealing with global data? Or are you concerned about the impact on first execution?

Reply:
another possiblity if we want to do this at import time could be to add a new module with those definitions and only import them if numpy is present